### ISSUE TITLE
Passing variables to delegate constuctors should update their state to non-nullable

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#38171]
**Version Used**: 

```
commit aef5f7afb9208964afe7d0a2b4536b735649b347 (HEAD -> release/dev16.4-preview1, origin/release/dev16.4-preview1)
Merge: 7c5e8fdf0b c4895b82cd
Author: Petr Hou<C5><A1>ka <t-pehouk@microsoft.com>
Date:   Tue Aug 20 23:33:36 2019 -0700

    Merge pull request #37884 from petrroll/move-refa-to-helpers6

    Move some ComputeRefactoringsAsync to common helpers 6/n
```

**Steps to Reproduce**:

```cs
#nullable enable
using System;

class C
{
  static void M(Action? a)
  {
    var d = new D(a); // CS8601: Possible null reference assignment; correct
    a(); // CS8602: Dereference of a possibly null reference; incorrect
  }
}

delegate void D();
```

**Expected Behavior**:
No warnings for `a()`. If `a` was null an exception would've been thrown at `new D(a)`.

**Actual Behavior**:
`CS8602: Dereference of a possibly null reference` is reported for `a()`

I believe updating the argument's slot state was forgotten when fixing https://github.com/dotnet/roslyn/issues/37984. /cc @RikkiGibson

**Comment by RikkiGibson** [comment]
Hmm, maybe there’s a better method to use in NullableWalker to report the diagnostic which updates the argument’s flow state.
